pkg/bootstrap/platform/gcp.go

```diff
@@ -104,11 +110,29 @@ var (
 		return fmt.Sprintf("https://container.googleapis.com/v1/projects/%s/locations/%s/clusters/%s",
 			projectID, clusterLocation, clusterName), nil
 	}
+
+	isMetadataEndpointAccessibleFn = func() bool {
```

Comment:
> so the issue is that resolution is OK, but actual connection is blocked?

Yes.

> And the issue is that using something like metadata.Zone() as a proxy won't work because the default retry is too long?

The issue is that we have a series of requests that each takes about 10 seconds because the connection is not possible,

With the change we discover that the connection is not possible early on, and we skip all those attempts.